eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/mutable/primitive/BooleanHashBag.java

```diff
@@ -611,6 +612,40 @@ public MutableBooleanBag reject(BooleanPredicate predicate)
         return result;
     }
 
+    @Override
+    public RichIterable<BooleanIterable> chunk(int size)
+    {
+        if (size <= 0)
+        {
+            throw new IllegalArgumentException("Size for groups must be positive but was: " + size);
+        }
+        if (this.isEmpty())
```

Comment:
Same comment as previous... this can be simplified initializing first, then collapsing the if statement to checking for notEmpty.